common/thorhelper/roxiehelper.cpp

```diff
@@ -589,10 +590,52 @@ bool CSafeSocket::readBlock(StringBuffer &ret, unsigned timeout, HttpHelper *pHt
             else
                 left = len = 0;
 
-            pHttpHelper->setIsHttp(true);
             if (!len)
                 throw MakeStringException(THORHELPER_DATA_ERROR, "Badly formed HTTP header");
         }
+        else if (pHttpHelper != NULL && strncmp((char *)&len, "GET", 3) == 0)
+        {
+#define MAX_HTTP_GET_HEADERSIZE 16000
```

Comment:
I'll see about getting rid of the limit entirely.  It was copied from HTTP-POST above but increased arbitrarily because REST users like to create some pretty complex URLs.  Better not to have it.